app/models/user_track.rb

```diff
@@ -55,9 +55,10 @@ def num_available_side_exercises
                 count
       +
       track.exercises.side.
-                where(unlocked_by: nil).
-                where.not(id: completed_side_exercise_ids).
-                count
+                      active.
+                      where(unlocked_by: nil).
```

Comment:
Also, can you suggest a way to collapse the scopes into a concept? As I going through this code, I could understand the scopes, but it was confusing on what combination of scopes to use.

Reply:
So we could add a scope for `bonus` which would be `side.where(unlocked_by: nil)`. However, I'd be a little worried that we would (in the future) think that `bonus` and `side` are mutually exclusive, and so double-up. If we only added bonus to be `where(unlocked_by: nil)` I'd be concerned that we'd (in the future) think that `bonus` only included `side` exercises and end up with core in there. So I don't know what the right thing to do is.

Reply:
As this part is already relatively stable and not needing anymore changes and since we're unsure ourselves, I'll just leave this as it is and not refactor it.

With our discussion on changing the tracks last night, I'm positive that we will arrive on better formed definitions of what the types of exercises are.